provider/resources.go

```diff
@@ -1776,7 +1777,45 @@ func Provider() tfbridge.ProviderInfo {
 
 			// CloudRun Resources
 			"google_cloud_run_domain_mapping": {Tok: gcpResource(gcpCloudRun, "DomainMapping")},
-			"google_cloud_run_service":        {Tok: gcpResource(gcpCloudRun, "Service"), Fields: nameField(lowercaseAutoName())},
+			"google_cloud_run_service": {
+				Tok: gcpResource(gcpCloudRun, "Service"),
+				// // PreCheckCallback: func(
+				// // 	ctx context.Context, config resource.PropertyMap, meta resource.PropertyMap,
+				// // ) (resource.PropertyMap, error) {
+				// // 	panic("I am your provider")
+				// // },
+				// Fields: map[string]*tfbridge.SchemaInfo{
+				// 	"name": tfbridge.AutoNameWithCustomOptions("name",
+				// 		// Name is auto-named without any suffix.
+				// 		tfbridge.AutoNameOptions{Randlen: 0}),
+				// 	"autogenerate_revision_name": {
+				// 		Default: &tfbridge.DefaultInfo{
+				// 			Value: true,
+				// 		},
+				// 	},
+				// },
+				Fields: nameField(lowercaseAutoName()),
+				TransformFromState: func(ctx context.Context, state resource.PropertyMap) (resource.PropertyMap, error) {
+					jsonBytes, err := json.Marshal(state)
+					if err != nil {
+						return nil, err
+					}
+					log.Printf("state: %s", string(jsonBytes))
+					if _, md := state["metadata"]; md {
+						metadata := state["metadata"].ObjectValue()
```

Comment:
This will panic if the property value is not an object. You need to check `state["metadata"].IsObject()` first